src/sentry/flags/endpoints/secrets.py

```diff
@@ -83,16 +83,35 @@ def post(self, request: Request, organization: Organization) -> Response:
         if not validator.is_valid():
             return self.respond(validator.errors, status=400)
 
-        FlagWebHookSigningSecretModel.objects.create_or_update(
-            organization=organization,
-            provider=validator.validated_data["provider"],
-            values={
-                "created_by": request.user.id,
-                "date_added": datetime.now(tz=timezone.utc),
-                "provider": validator.validated_data["provider"],
-                "secret": validator.validated_data["secret"],
-            },
-        )
+        # these scopes can always update or post secrets
+        has_update_or_post_access = request.access.has_scope(
+            "org:write"
+        ) or request.access.has_scope("org:admin")
+
+        try:
+            secret = FlagWebHookSigningSecretModel.objects.filter(
+                organization_id=organization.id
+            ).get(provider=validator.validated_data["provider"])
+            # allow update access if the user created the secret
+            if request.user.id == secret.created_by:
+                has_update_or_post_access = True
+        except FlagWebHookSigningSecretModel.DoesNotExist:
+            # anyone can post a new secret
+            has_update_or_post_access = True
+
+        if has_update_or_post_access:
+            FlagWebHookSigningSecretModel.objects.create_or_update(
+                organization=organization,
+                provider=validator.validated_data["provider"],
+                values={
+                    "created_by": request.user.id,
+                    "date_added": datetime.now(tz=timezone.utc),
+                    "provider": validator.validated_data["provider"],
+                    "secret": validator.validated_data["secret"],
+                },
+            )
+        else:
+            return Response("Not authorized.", status=401)
```

Comment:
The user _is_ authorized at this point, and has at least these org-level scopes:
https://github.com/getsentry/sentry/blob/e8371a7e81a05fdae24eec4d5ed042a536697cbf/src/sentry/api/bases/organization.py#L236
I think 403 (Forbidden) is more appropriate than 401 in this case. We can also add some more details why the request is forbidden:
```suggestion
            return Response("You must be an organization owner, manager, or creator of this secret in order to perform this action.", status=403)
```

Reply:
gotcha, updated the response